### PR TITLE
use NFD for node labelling

### PR DESCRIPTION
**Pull request by pohly**
By using NFD we ensure that the integration with it works as intended.

TODO:
- [X] documentation
- [X] clean up commits (split out removal of Clear Linux workaround)
- [x] ensure that downgrade tests work - they probably don't, because the node labels prevent running the older version

Fixes: issue #747

### CODE REVIEW

**Comment by pohly** [comment]
@avalluri I tested manually that version skew testing still passes. This is ready for review.

@marquiz FYI, we are going to use and recommend NFD for node labeling - this has been long overdue :sweat_smile:

**Comment by marquiz** [comment]
> FYI, we are going to use and recommend NFD for node labeling - this has been long overdue

Good to get some new users onboard 👍 Especially users we're aware of 😄 Please don't hesitate to bring up any issues or wishes for improvement you might have.

File: docs/autotest.md [hunk 1]
```diff
@@ -56,8 +56,11 @@ virtual machines.
 The first node is the Kubernetes master without
 persistent memory.
 The other three nodes are worker nodes with one emulated 32GB NVDIMM each.
-After the cluster has been formed, `make start` adds `storage=pmem` label
-to the worker nodes and deploys the PMEM-CSI driver.
+After the cluster has been formed, `make start` adds NFD to label
```
Comment (avalluri) [inline docs/autotest.md:59]: 'adds NFD' sounds not appropriate to me, what about something line 'starts NFD'? And a link to NFD project seems appropriate here.

File: docs/install.md [hunk 2]
```diff
@@ -209,10 +209,25 @@ on the Kubernetes version.
 
 - **Label the cluster nodes that provide persistent memory device(s)**
 
+The local part of PMEM-CSI runs on all nodes with a certain label. For
```
Comment (avalluri) [inline docs/install.md:212]: @pohly What exactly 'local' refers to here?

File: docs/autotest.md (hunk 1 above)
Comment (avalluri) [reply docs/autotest.md:59]: Okay, a later part of the documentation added the link to NFD, so it is fine to omit it here.

File: docs/install.md [hunk 3]
```diff
@@ -448,13 +463,16 @@ verify that the node labels have been configured correctly
...
+The command output must indicate that every node with PMEM has at least two labels:
 ``` console
 pmem-csi.intel.com/node=<NODE-NAME>,storage=pmem
 ```
 
-If **storage=pmem** is missing, label manually as described above. If
-**pmem-csi.intel.com/node** is missing, then double-check that the
+**storage=pmem** is the label that has to be added manually as
+described above.  When using NFD, the node should have the
+`feature.node.kubernetes.io/memory-nv.dax: true` label.
```
Comment (avalluri) [inline docs/install.md:473]: Replacing ';' with '=' is most appropriate here.

File: docs/install.md (hunk 2 above)
Comment (pohly) [reply docs/install.md:212]: The node server.

Saying the "The node part of PMEM-CSI runs on all nodes with a certain label" felt odd.

File: docs/autotest.md (hunk 1 above)
Comment (pohly) [reply docs/autotest.md:59]: "`make start` starts NFD" stutters.

How about "`make start` installs NFD"?

Comment (avalluri) [reply docs/autotest.md:59]: > How about "`make start` installs NFD"?

Sounds good.

File: docs/install.md (hunk 2 above)
Comment (pohly) [reply docs/install.md:212]: Replaced with "PMEM-CSI manages PMEM on those nodes that have a certain label."

File: docs/install.md (hunk 3 above)
Comment (pohly) [reply docs/install.md:473]: Fixed.

**Review by avalluri — APPROVED** [review]

**Pull request merged**